runtimevar/etcdvar/etcdvar.go

```diff
@@ -97,7 +97,7 @@ func (w *watcher) WatchVariable(ctx context.Context, _ driver.State) (driver.Sta
 // It always return s.
 func (w *watcher) updateState(s, prev *state) *state {
 	if s.err != nil && prev != nil && prev.err != nil {
-		if s.err == prev.err {
+		if s.err == prev.err || s.err.Error() == prev.err.Error() {
```

Comment:
Does `s.err.Error() == prev.err.Error()` include `s.err == prev.err`?

Reply:
Yes, but it's less efficient.